kpops/pipeline_generator/pipeline.py

```diff
@@ -155,44 +145,29 @@ def parse_components(self, component_list: list[dict]) -> None:
                 else:
                     raise ParsingException() from ex
 
-    def populate_pipeline_component_names(
-        self, inflated_components: list[PipelineComponent]
-    ) -> None:
-        for component in inflated_components:
-            component.name = component.prefix + component.name
-            with suppress(
-                AttributeError  # Some components like Kafka Connect do not have a name_override attribute
-            ):
-                if component.app and getattr(component.app, "name_override") is None:
-                    setattr(component.app, "name_override", component.name)
-
     def apply_component(
-        self,
-        component_data: dict,
-        component_class: type[PipelineComponent],
-        previous_component: PipelineComponent | None,
-    ) -> list[PipelineComponent]:
+        self, component_class: type[PipelineComponent], component_data: dict
+    ) -> None:
         component = component_class(
             config=self.config,
             handlers=self.handlers,
             **component_data,
         )
         component = self.enrich_component(component)
-        # weave from topics
-        if previous_component and previous_component.to:
-            component.weave_from_topics(previous_component.to)
 
         # inflate & enrich components
```

Comment:
Maybe extract in a method and remove the comment :)

Reply:
I sympathize with the motive, in this case I suppose it's actually easier to understand if we leave the logic in `apply_component`, because it goes through several steps (enrich base component, inflate, enrich inflated components, weave from topics, finally add components to component index). otherwise it gets pretty fragmented and hard to name each of the methods :D